Yoast/javascript
# P2-505 Adds required blocks to sidebar

## Summary


This PR can be summarized in the following changelog entry:

* Introduces a panel in the sidebar containing the required blocks with the possibility to add a block that isn't added yet.

## Relevant technical choices:

* At the moment the block doesn't contain the checkmark icon for the blocks that are added already. The reason for this is that a decision should be made on which icon to use for that.

## Test instructions

This PR can be tested by following these steps:

enable the schema blocks feature by adding the following line to loader.php below the use statements:
`\define( 'YOAST_SEO_SCHEMA_BLOCKS', true );`
* start a new post
* show the sidebar
* add a recipe block
* see the "paragraph" block in the sidebar; press add
* see there's a new paragraph in the post; the sidebar says OK
* remove the paragraph block; the sidebar allows you to add again.

the required blocks are goverened by recipe.block.php. you can play around with this a little but it's quite technical:
```{{inner-blocks required-blocks=[ { "name": "core/paragraph", "option": "One" } ] ```
```{{inner-blocks required-blocks=[ { "name": "core/paragraph", "option": "One" }, { "name": "core/image", "option": "Multiple" } ] ```

* Link the monorepo, do a `yarn build` in the schema templates package and do a `grunt build` in wordpress-seo

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * the feature is still disabled and should not have any impact on existing code.

## UI changes

## Quality assurance


Fixes #
